src/StoreApi/Utilities/NoticeHandler.php

```diff
@@ -1,31 +1,33 @@
 <?php
 namespace Automattic\WooCommerce\Blocks\StoreApi\Utilities;
 
-use Automattic\WooCommerce\Blocks\StoreApi\Exceptions\RouteException;
+use WP_Error;
 
 /**
  * NoticeHandler class.
- * Helper class to convert notices to exceptions.
+ * Helper class to convert notices to wp errors.
  *
  * @internal This API is used internally by Blocks--it is still in flux and may be subject to revisions.
  */
 class NoticeHandler {
 
 	/**
-	 * Convert queued error notices into an exception.
+	 * Collects queued error notices into a \WP_Error.
 	 *
 	 * For example, Payment methods may add error notices during validate_fields call to prevent checkout.
-	 * Since we're not rendering notices at all, we need to convert them to exceptions.
+	 * Since we're not rendering notices at all, we need to catch them and group them in a single WP_Error instance.
 	 *
-	 * This method will find the first error message and thrown an exception instead. Discards notices once complete.
-	 *
-	 * @throws RouteException If an error notice is detected, Exception is thrown.
+	 * This method will discards notices once complete.
 	 *
 	 * @param string $error_code Error code for the thrown exceptions.
+	 *
+	 * @return \WP_Error The WP_Error object containing all error notices.
 	 */
-	public static function convert_notices_to_exceptions( $error_code = 'unknown_server_error' ) {
```

Comment:
There are other consumers of this such as legacy payments API. This needs to stay in place (tests too), with the wp_error version alongside.

Reply:
https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/8c7c9009a1a7cd52165d6d1a381c7c0607649f5b/src/Payments/Api.php#L152

Reply:
Oops! Missed that! Reverting...